File: Semestrul 4/MPP/lab11/java/TriathlonNetworking/src/main/java/triathlon/network/dto/DTOUtils.java
package triathlon.network.dto;

import triathlon.model.*;
import java.util.ArrayList;
import java.util.List;

public class DTOUtils {

    public static RefereeDTO fromReferee(Referee referee) {
        if (referee == null) return null;
        return new RefereeDTO(
                referee.getId(),
                referee.getUsername(),
                referee.getPassword()
        );
    }

    public static Referee toReferee(RefereeDTO dto) {
        if (dto == null) return null;
        Referee referee = new Referee(
                dto.getUsername(),
                dto.getPassword(),
                "", 
                "", 
                ""  
        );
        referee.setId(dto.getId());
        return referee;
    }

    public static ResultDTO fromResult(Result result) {
        if (result == null) return null;
        return new ResultDTO(
                result.getId(),
                result.getEvent().getId(),
                result.getParticipant().getId(),
                result.getDiscipline(),
                result.getPoints()
        );
    }

    public static List<ResultDTO> fromResultList(List<Result> results) {
        List<ResultDTO> dtos = new ArrayList<>();
        for (Result result : results) {
            dtos.add(fromResult(result));
        }
        return dtos;
    }
}

File: Semestrul 4/MPP/lab09-10/server .net/TriathlonNetworking/DTOs/DTOUtils.cs
using System.Collections.Generic;
using Triathlon.Model;

namespace Triathlon.Network.Dto
{
    public static class DTOUtils
    {
        public static RefereeDTO? FromReferee(Referee referee)
        {
            if (referee == null) return null;
            return new RefereeDTO(
                referee.Id,
                referee.Username,
                referee.Password
            );
        }

        public static Referee? ToReferee(RefereeDTO dto)
        {
            if (dto == null) return null;
            Referee referee = new Referee(
                dto.Username,
                dto.Password,
                "", // email
                "", // firstName
                ""  // lastName
            );
            referee.Id = dto.Id;
            return referee;
        }

        public static ResultDTO FromResult(Result result)
        {
            if (result == null) return null;
            return new ResultDTO(
                result.Id,
                result.Event.Id,
                result.Participant.Id,
                result.Discipline,
                result.Points
            );
        }

        public static List<ResultDTO> FromResultList(List<Result> results)
        {
            List<ResultDTO> dtos = new List<ResultDTO>();
            foreach (Result result in results)
            {
                dtos.Add(FromResult(result));
            }
            return dtos;
        }
    }
}